sql/parser/eval_test.go

```diff
@@ -233,21 +233,22 @@ func TestEval(t *testing.T) {
 		// Cast expressions.
 		{`true::boolean`, `true`},
 		{`true::int`, `1`},
-		{`true::float`, `1`},
+		{`true::float`, `1.0`},
 		{`length(true::text)`, `4`},
 		{`false::boolean`, `false`},
 		{`false::int`, `0`},
-		{`false::float`, `0`},
+		{`false::float`, `0.0`},
 		{`length(false::text)`, `5`},
 		{`1::boolean`, `true`},
 		{`0::boolean`, `false`},
 		{`1::int`, `1`},
-		{`1::float`, `1`},
+		{`1::float`, `1.0`},
 		{`length(123::text)`, `3`},
 		{`1.1::boolean`, `true`},
 		{`0.0::boolean`, `false`},
 		{`1.1::int`, `1`},
 		{`1.1::float`, `1.1`},
+		{`1000000::float`, `1000000.0`},
```

Comment:
Is TestParse the correct place for this test? It doesn't seem to be printing the number using the affected code. 
For example, both of these cases pass:
> {`SELECT 1000000.0`},
> {`SELECT 1e+06`},

Alternatively, in TestEval, we can do:
> {`999999::float`, `999999.0`},
> {`1000000::float`, `1e+06`},
Or, if you prefer:
> {`999999.0`, `999999.0`},
> {`1000000.0`, `1e+06`},